nix/module/server.nix

```
@@ -0,0 +1,111 @@
+{ pkgs, lib, config, ... }:
+
+{
+  options.services.filetracker = {
+    enable = lib.mkEnableOption "filetracker server";
+
+    package = lib.mkPackageOption pkgs "Filetracker" {
+      default = [ "python38Packages" "filetracker" ];
+    };
+
+    listenAddress = lib.mkOption {
+      default = "0.0.0.0";
+      description = "The address that filetracker-server will listen on";
+      type = lib.types.str;
+    };
+
+    port = lib.mkOption {
+      default = 9999;
+      description = "The port that filetracker-server will listen on";
+      type = lib.types.port;
+    };
+
+    openFirewall = lib.mkOption {
+      default = true;
+      description = "Whether to open the provided port in the firewall";
+      type = lib.types.bool;
+    };
+
+    ensureFiles = lib.mkOption {
+      default = { };
+      description = "Files that should be added to filetracker after start";
+      type = lib.types.attrsOf lib.types.path;
+    };
+
+    workers = lib.mkOption {
+      default = "auto";
+      description = "The number of gunicorn workers to spawn";
+      type = with lib.types; oneOf [ (strMatching "auto") ints.positive ];
+    };
+  };
+
+  config =
+    let
+      cfg = config.services.filetracker;
+      python = cfg.package.pythonModule;
+
+      createEnsureService = remotePath: localPath:
+        let
+          systemdEscapedPath = builtins.replaceStrings [ "/" ] [ "-" ] (lib.removePrefix "/" remotePath);
+          serviceName = "filetracker-put-${systemdEscapedPath}";
```

Theoretically nixos already has a function that does this properly [here](https://github.com/NixOS/nixpkgs/blob/e5addb65b062a00de864446d083ecdb141999eb6/nixos/lib/utils.nix#L46) but I think we'd have to import this file manually for this to work.
```suggestion
          serviceName = "filetracker-put-${escapeSystemdPath remotePath}";
```
(this suggestion only works if this function is imported and added to scope)

---

Imo, let's leave this for now

---

Lets at least add a FIXME for this.
```suggestion
          // FIXME: This may not be entirely correct, a better alternative would be using escapeSystemdPath from nixpkgs/nixos/lib/utils.nix or at least copying their solution.
          systemdEscapedPath = builtins.replaceStrings [ "/" ] [ "-" ] (lib.removePrefix "/" remotePath);
          serviceName = "filetracker-put-${systemdEscapedPath}";
```